footil/date.py

```diff
@@ -123,3 +123,20 @@ def get_last_day_date_str(
     date_fmt = to_date(dt, fmt=fmt)
     modified_dt = get_last_day_date(date_fmt, months=months)
     return to_string(modified_dt, fmt=new_fmt)
+
+
+def _check_time_str(dt: str, fmt: str) -> None:
+    try:
+        datetime.datetime.strptime(dt, fmt)
+    except ValueError as e:
+        raise e
```

Comment:
There is no point to catch and raise same error (you do nothing here).
You can just do:

```python
def _check_time_str(dt: str, fmt: str) -> None:
    datetime.datetime.strptime(dt, fmt)
```

And then you can actually return it and then it is reduced to same thing..

So what we actually need is this (first one already exists), `to_date` and `to_datetime` functions. So we dont need explicit "check" here. It will be done while converting.

Also `to_datetime`, should probably go to new module called datetime.py, to not be mixed with date stuff (and then we need to move `DATETIME_FMT` there too. But need to make sure it is not used anywhere already?).